### DISCUSSION TITLE
カスタムパーサーの使い方について

### DISCUSSION THREAD

**Discussion by chaftta** [kyubuns/AkyuiUnity#22] [Q&A]
・カスタムパーサーを使いたいのですがデフォルトパーサーの処理とコンフリクトが起きているようなのでデフォルトパーサーを無効化する方法はないでしょうか？
・パーサーやトリガーにプライオリティを付けて実行される順番を制御する方法は、ないでしょうか？

**Comment by chaftta** [comment]
もうしわけありません、パーサーのコンフリクトは勘違いでした

**Comment by kyubuns** [comment]
質問ありがとうございます！
パーサーやトリガーは、それぞれリストの上から順番に実行されることが保証されています。
（UI的に並び替えがしにくくて申し訳ないのですが）

例えば以下の設定なら、SetupAuto9Slicer -> SetupAnKuchen -> TextMeshPro -> CreateSpriteAtlas の順番に実行されます。
<img width="416" alt="Screen Shot 2021-05-20 at 17 59 12" src="https://user-images.githubusercontent.com/961165/118950357-1647c500-b995-11eb-85e4-2a3f80e7344d.png">

**Reply by chaftta** [reply]
ご回答ありがとうございます
並び替えすればトリガーの順序は解決致しました
ありがとうございます！

質問ばかりで恐縮ですが、スクロール表示の変換でちょっと悩んでいます

![Image](https://user-images.githubusercontent.com/84495634/118964899-fcad7a00-b9a2-11eb-9075-8a67090f463a.png)

デザイナにXDのスクロールで良く分からないグループを一段入れないといけないのは嫌な感じと言われたので検討中です
何かアドバイスがありましたら宜しくお願い致します

**Reply by kyubuns** [reply]
すみません、質問の意図を正しく認識できているか自身がないのですが、
XD上で、リストっぽくItemの要素がたくさん並んだプレビューを見る必要がないのなら、
↓のようにScroll直下にItemを配置しても問題ありません。
<img width="712" alt="Screen Shot 2021-05-20 at 20 35 17" src="https://user-images.githubusercontent.com/961165/118971892-e6a3b780-b9aa-11eb-88f4-c68a1b3065dd.png">

**Reply by chaftta** [reply]
実行時に、ItemのコピーインスタンスをItemsに沢山追加する感じを想定しております

**Reply by kyubuns** [reply]
話をまとめますと、

デザイナーさんに↓のように、無駄なグループの無いXDを作っていただくと、
<img width="712" alt="118971892-e6a3b780-b9aa-11eb-88f4-c68a1b3065dd" src="https://user-images.githubusercontent.com/961165/118978560-739e3f00-b9b2-11eb-9f3d-1a176c5b0d99.png">

上げて頂いた画像の、「現状のプレハブ」の状態にコンバートされます。
![118964899-fcad7a00-b9a2-11eb-9075-8a67090f463a](https://user-images.githubusercontent.com/961165/118978677-89abff80-b9b2-11eb-8222-6bb5cb23a940.png)

この時、Unity上の"Scroll"に、XD上の"Scroll"グループの状態から、マージンなどが自動的に設定されている状態になります。
実行時には、Itemをコピーしたインスタンスを（Itemsではなく）Content以下に直接生成してもらえれば想定した挙動になるのではないかと思います。

**Reply by chaftta** [reply]
ありがとうございます！
試してみます
